code/controllers/subsystem/pathfinding.dm

```diff
@@ -0,0 +1,379 @@
+// This file contains the pathfinding subsystem and all base procs related to pathfinding.
+
+SUBSYSTEM_DEF(pathfinding)
+	name = "Pathfinding"
+	init_order = INIT_ORDER_PATH
+	flags = SS_NO_FIRE
+	var/list/queues = list()
```

Comment:
new queue system, not sure if better/worse than flowcache but i didn't like the fact that so many helper datums were being used when lists could just work.